pkg/requirements/requirements.go

```diff
@@ -107,6 +114,29 @@ func AddUserPasswordToURLFromDir(gitURL, path string) (string, error) {
 	return gitURL, nil
 }
 
+// GitCredsFromCluster processes the Git credentials from the secret mount path and updates the gitURL accordingly.
+func GitCredsFromCluster(gitURL string) (string, error) {
+	secretMountPath := os.Getenv(credentialhelper.GIT_SECRET_MOUNT_PATH)
+	if secretMountPath != "" {
+		err := credentialhelper.WriteGitCredentialFromSecretMount()
+		if err != nil {
+			return "", fmt.Errorf("failed to write git credentials file for secret %s : %w", secretMountPath, err)
+		}
+
+		gitURL, err = AddUserPasswordToURLFromDir(gitURL, secretMountPath)
+		if err != nil {
+			return "", fmt.Errorf("failed to add username and password to git URL: %w", err)
+		}
+	} else {
+		if kube.IsInCluster() {
+			log.Logger().Warnf("no $GIT_SECRET_MOUNT_PATH environment variable set")
+		} else {
+			log.Logger().Debugf("no $GIT_SECRET_MOUNT_PATH environment variable set")
+		}
+	}
+	return gitURL, nil
+}
```

Comment:
Think this can be private. No reason for anyone outside of this package to use it.

Functions are made private in Go by having a lowercase letter at the start of the function name; so in this case:
```go
func gitCredsFromCluster(gitURL string) (string, error)
```

Reply:
Done